lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4023,6 +4024,90 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
                              transValue(II->getArgOperand(0), BB),
                              transValue(II->getArgOperand(1), BB), BB);
   }
+  case Intrinsic::vector_reduce_add:
+  case Intrinsic::vector_reduce_mul:
+  case Intrinsic::vector_reduce_and:
+  case Intrinsic::vector_reduce_or:
+  case Intrinsic::vector_reduce_xor: {
+    Op Op;
+    if (IID == Intrinsic::vector_reduce_add) {
+      Op = OpIAdd;
+    } else if (IID == Intrinsic::vector_reduce_mul) {
+      Op = OpIMul;
+    } else if (IID == Intrinsic::vector_reduce_and) {
+      Op = OpBitwiseAnd;
+    } else if (IID == Intrinsic::vector_reduce_or) {
+      Op = OpBitwiseOr;
+    } else {
+      Op = OpBitwiseXor;
+    }
+    VectorType *VT = cast<VectorType>(II->getArgOperand(0)->getType());
+    SPIRVValue *SV = transValue(II->getArgOperand(0), BB);
+    SPIRVTypeInt *ResultType =
+        BM->addIntegerType(VT->getElementType()->getIntegerBitWidth());
```

Comment:
Nit: 
```suggestion
        transType(VT->getElementType());
```

Reply:
Hmm... for me it looks more hacky and slower than just adding a IntegerType. BM->addIntegerType explictly is telling what's is doing - adding a necessary type (imo. should be something like getOrCreateIntegerType).
transType here is going around  by passing through whole translation logic to achieve the same goal without explicitly telling intention.
In addition to that transType is returning SPIRVType * but addIntegerType returns SPIRVIntegerType * which have more information than it's parent type. 
We can of course remove addXXXType api and establish that's transType it the way - it would be good to have discussion with community about that.

Reply:
Not a blocker. So marking this resolved.